prov/gni/src/gnix_mr.c

```diff
@@ -399,13 +415,252 @@ static int __gnix_destruct_registration(void *context)
 	return GNI_RC_SUCCESS;
 }
 
+
+#if HAVE_UDREG
+void *__udreg_register(void *addr, uint64_t length, void *context)
+{
+	struct gnix_fid_mem_desc *md;
+	struct gnix_fid_domain *domain;
+
+	domain = (struct gnix_fid_domain *) context;
+
+    /* Allocate an udreg info block for this registration. */
+    md = malloc(sizeof(struct gnix_fid_mem_desc));
+    if (!md) {
+	GNIX_WARN(FI_LOG_MR, "failed to allocate memory for registration\n");
+	return NULL;
+    }
+
+    return __gnix_generic_register(domain, md, addr, length, NULL,
+		GNI_MEM_READWRITE, -1);
+}
+
+
+uint32_t __udreg_deregister(void *registration, void *context)
+{
+	gni_return_t grc;
+
+	grc = __gnix_deregister_region(registration, NULL);
+
+	return (grc == GNI_RC_SUCCESS) ? 0 : 1;
+}
+
+
+/* Called via dreg when a cache is destroyed. */
+void __udreg_cache_destructor(void *context)
+{
+    /*  Nothing needed here. */
+}
+
+static int __udreg_init(struct gnix_fid_domain *domain)
+{
+	udreg_return_t urc;
+
+	udreg_cache_attr_t udreg_cache_attr = {
+		.cache_name =           {"gnix_app_cache"},
+		.max_entries =          4096,
```

Comment:
Is there any reason to hard code this?  I think we should have a default value that can be changed via domain open_ops.